### PR TITLE
Add blank canvas widget with search functionality and update tile ren…

### PR DESCRIPTION
**Pull request by olivernosto**
Introduce blank canvas & improve settings

## Checklist

- [ ] Pull Request is properly described.
- [ ] The corresponding Jira ticket is updated.
- [ ] I have requested a review from at least one reviewer.
- [ ] The changes are tested
- [ ] I have verified my UX changes with a designer
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: src/libs/tile.handlers.ts [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
 import { renderHTMLWithTemplates, renderTilesWithTemplate } from "@stackla/handlebars"
 
-export function getTilesToRender(page: number, limit: number) {
+export function getTilesToRender(req: Request) {
+  const page = (req.query.page ?? 0) as number
+  const limit = (req.query.limit ?? 25) as number
   const startIndex = limit * (page - 1)
   const endIndex = limit * page
-  return tiles.slice(startIndex, endIndex)
+  const mutatedTiles = tiles.slice(startIndex, endIndex)
```
Comment (olivernosto) [inline src/libs/tile.handlers.ts:12]: This is just a sample method that allows for search simulation

File: widgets/slider/utils.ts [hunk 2]
```diff
@@ -17,8 +17,8 @@ export function getTileSizeUnitless(settings: Features["tileSizeSettings"]) {
  * @returns the gap value for slider inline tile grid
  */
 export function inlineTileGap() {
-  const { inline_tile_margin } = sdk.getStyleConfig()
-  const value = Number(inline_tile_margin)
+  const { margin } = sdk.getStyleConfig()
+  const value = Number(margin)
```
Comment (olivernosto) [inline widgets/slider/utils.ts:21]: inline tile margin does not exist

File: widgets/blankcanvas/config.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import { ExpandedTileOptions, InlineTileOptions, Style } from "packages/widget-utils/dist/esm"
+
+export const config: {
+  style: Style
```
Comment (olivernosto) [inline widgets/blankcanvas/config.ts:4]: Blank canvas does not have these settings in the backend, so the user defines them from the frontend

File: src/libs/express.ts [hunk 4]
```diff
@@ -123,7 +123,7 @@ export async function getContent(widgetType: string, retry = 0): Promise<Preview
   }
 }
 
-async function getHTML(content: PreviewContent, page: number = 1, limit: number = 25) {
+async function getHTML(content: PreviewContent, request) {
```
Comment (manikandan-ravikumar) [inline src/libs/express.ts:126]: No type on the second parameter?

File: src/libs/tile.handlers.ts [hunk 5]
```diff
@@ -1,25 +1,44 @@
...
+        }
+        return tileField.toLowerCase().includes(search.toLowerCase())
+      })
+    })
+  }
+  return mutatedTiles
 }
 
-export async function getAndRenderTiles(widgetContainer: IDraftRequest, page: number = 1, limit: number = 25) {
+export async function getAndRenderTiles(widgetContainer: IDraftRequest, request) {
```
Comment (manikandan-ravikumar) [inline src/libs/tile.handlers.ts:30]: No type parameter for request?

File: src/libs/tile.handlers.ts [hunk 6]
```diff
@@ -1,25 +1,44 @@
...
   const tileTemplate = widgetContainer.customTemplates.tile.template
-  return renderTilesWithTemplate(tileTemplate, getTilesToRender(page, limit), widgetContainer.widgetOptions)
+  return renderTilesWithTemplate(tileTemplate, getTilesToRender(request), widgetContainer.widgetOptions)
 }
 
-export async function renderTemplates(widgetContainer: IDraftRequest, page: number = 1, limit: number = 25) {
+export async function renderTemplates(widgetContainer: IDraftRequest, request) {
```
Comment (manikandan-ravikumar) [inline src/libs/tile.handlers.ts:35]: No type parameter for request?

File: cypress/snapshots/base/cypress/e2e/widget/carousel.cy.js/carousel-tile.png
Comment (manikandan-ravikumar) [inline cypress/snapshots/base/cypress/e2e/widget/carousel.cy.js/carousel-tile.png:1]: Why was this snapshot updated for an unrelated change?

Comment (olivernosto) [reply cypress/snapshots/base/cypress/e2e/widget/carousel.cy.js/carousel-tile.png:1]: Removed.
